app/src/main/java/com/mapbox/memleak/MapScreen.kt

```diff
@@ -81,33 +75,43 @@ fun MapScreen(navigate: () -> Unit) {
                   .align(Alignment.TopStart)
             )
          },
-         mapInitOptionsFactory = { context ->
-            MapInitOptions(
-               context = context,
-               styleUri = "mapbox://styles/basebeta/clm3fw4kx00tk01qyco2x8f3p",
-               cameraOptions = CameraOptions.Builder()
-                  .center(
-                     Point.fromLngLat(
-                        cameraState.center.longitude,
-                        cameraState.center.latitude
-                     )
+         style = {
+            MapStyle(style = "mapbox://styles/basebeta/clm3fw4kx00tk01qyco2x8f3p")
+         },
+         mapViewportState = MapViewportState().apply {
+            setCameraOptions {
+               center(
+                  Point.fromLngLat(
+                     cameraState.center.longitude,
+                     cameraState.center.latitude
                   )
-                  .zoom(cameraState.zoom)
-                  .pitch(cameraState.pitch)
-                  .bearing(cameraState.bearing)
-                  .build(),
-            )
+               )
+               zoom(cameraState.zoom)
+               pitch(cameraState.pitch)
+               bearing(cameraState.bearing)
+            }
          }
       ) {
          val isMarkerVisible = remember { mutableStateOf(false) }
 
+         val exits: List<MapExit> = remember { json.decodeFromString(exitJSON) }
+         val options = remember {
+            exits.map {
+               PointAnnotationOptions()
+                  .withPoint(Point.fromLngLat(it.longitude, it.latitude))
+                  .withIconImage("blue-pin")
+            }
+         }
+
          // add pointannotation
-         val option = remember { PointAnnotationOptions()
-            .withPoint(Point.fromLngLat(-119.533632, 37.746011))
-            .withIconImage("blue-pin") }
+         val option = remember {
+            PointAnnotationOptions()
+               .withPoint(Point.fromLngLat(-119.533632, 37.746011))
+               .withIconImage("blue-pin")
+         }
 
          PointAnnotationGroup(
-            listOf(option),
+            options,
```

Comment:
I do see the point annotations to overlap if I enable iconAllowOverlap:
```suggestion
            options,
            iconAllowOverlap = true,
```
is it want you want to achieve?

Reply:
ah, I am an idiot. I had multiple PointAnnotationGroup's for the application this is in and neglected to add this field to one! My apologies